bftengine/include/bftengine/ClientMsgs.hpp

```diff
@@ -40,6 +40,7 @@ struct ClientRequestMsgHeader {
   uint64_t timeoutMilli;
   uint32_t cidLength = 0;
   uint16_t reqSignatureLength = 0;
+  uint32_t extraDataLength = 0;
```

Comment:
Will this make Client request message backward incompatible. We should check if this addition effects serialisation and deserialisation of this message.

Reply:
Actually this change doesn't affect the ClientRequestMsg. It's a mechanism to allocate extra buffer space if you are inheriting from the message (as it's done for PreProcess(ed)Result message).

The ClientRequestMsg message itself is not affected by this change because the extraDataLength is always set to 0 for it.

I also had concerns for serialisation/deserialisation so I added additional test cases in the message unit test. Check my changes in ClientRequestMsg_test.cpp/

Reply:
Checked the unit test, I think this should work.